### DISCUSSION TITLE
Affero General Public License

### DISCUSSION THREAD

**Discussion by IKhintsitskiy** [drakkan/sftpgo#342] [Q&A]
The project license was changed to Affero GPL a few days ago. I'm developing a proprietary software which runs in kubernetes cluster, I've included sftpgo as a separate docker container. My other containers communicate with sftpgo via REST, the end users of my product communicate with sftpgo via sftp.

How does the license change affect me?
Does Affero GPL have an "aggregation" exception, so that the proprietary code from my other containers communicating with SFTPgo is not required to be opensourced?

**Comment by drakkan** [comment]
Hi,

First of all you should consider a [sponsorship](https://github.com/sponsors/drakkan), maintaining SFTPGo takes a lot of effort and if I lose interest you could end up using unmaintained software within your product. This is obviously not required but much appreciated.

With traditional GPL, copyleft was associated with the concept of distribution of software, but you can run SFTPGo in the
cloud. AGPL fixes this "loophole" in GPL by saying that if you use the software over a network, you are bound by the copyleft.

To say this another way: if you modify the SFTPGo source code, the goal is that you have to contribute those modifications back and with the same license. 

Your application, even though it talks to the SFTPGo via the exposed REST API, is a separate program and "work" so it is not required to be open source. If, instead, you include SFTPGo inside your program, by linking it, it must be open source.

You also have to respect the conditions summarized [here](https://github.com/drakkan/sftpgo/blob/main/LICENSE).

If, for some reason, you need a different license you can contact me privately but I generally prefer open source usage, I have declined all relicensing requests received so far.

**Comment by wizardishungry** [comment]
@drakkan Don't forget to change the copyright from "FSF" to yourself.

**Comment by drakkan** [comment]
First of all I must say that I am not a lawyer and for this reason I asked for legal advice on the matter. Based on initial feedback from the legal consultation, I believe my old answer is not accurate. A correct interpretation of the AGPL is [this](https://min.io/compliance). I don't think SFTPGo can be included in proprietary products. When you include SFTPGo in your proprietary product, the semantics of the communication are intimate enough to consider your product a derivative work.

Please also note that SFTPGo is a registered trademark.

Thank you for your understanding and sorry for my previous inaccurate response.
